### ISSUE TITLE
Safari: no sound

### ISSUE THREAD

**Issue by johannes87** [d07RiV/diabloweb#52]
I don't have sound in Safari v12.1.2 (14607.3.9). It works in Chrome Version 76.0.3809.87.

**Comment by d07RiV** [comment]
I don't have Safari, but can you check the console for any errors? It should be accessible via F12.

**Comment by OrangBuangan** [comment]
What device do you use?

**Comment by MANassar** [comment]
I dont have sound on the iPad on any browser, tried Safari, Firefox, and Chrome. Not sure if its the same issue as the OP.

**Comment by datenreisender** [comment]
There is this error in the console that looks related: `[Error] TypeError: i.copyToChannel is not a function. (In 'i.copyToChannel(a.subarray(c*o,c*o+o),c)', 'i.copyToChannel' is undefined)
	create_sound `

**Comment by d07RiV** [comment]
There seems to be conflicting information about WebAudio availability in Safari. I also changed the way audio is loaded earlier today, maybe it would work better now.

**Comment by datenreisender** [comment]
With DiabloWeb v1.0.24 I the problem still exists on Safari. I currently see these errors in the console:
```
[Error] TypeError: Not enough arguments, decodeAudioData – sound.js:44
[Error] TypeError: i.copyToChannel is not a function. (In 'i.copyToChannel(a.subarray(c*o,c*o+o),c)', 'i.copyToChannel' is undefined), create_sound_raw – sound.js:32
```

**Comment by d07RiV** [comment]
Oh, there are indeed some things Safari doesn't support. I made a couple workarounds, try now (1.0.25).

**Comment by d07RiV** [comment]
Okay, try 1.0.26. Hope that's the last of them >.>

**Comment by datenreisender** [comment]
```
[Error] TypeError: function is not a constructor (evaluating 'new t(n,{pan:0})'), create_sound – sound.js:55
```

**Comment by d07RiV** [comment]
Could you enable "break on errors" and check what StereoPannerNode is equal to when it breaks on that line? That's really weird.

**Comment by datenreisender** [comment]
StereoPannerNode is undefined (`ReferenceError: Can't find variable: StereoPannerNode`) and `t` is equal to `window.webkitAudioPannerNode` which is just some native code, output in the console is just 
```
function webkitAudioPannerNode() {
    [native code]
} = $1
```

But I have no idea how to invoke `webkitAudioPannerNode`. All my tries just lead to an error and I found no documentation on the web. https://stackoverflow.com/a/54497389 describes an alternative way. I was not able to verify it completely but at least saw that `n.createPanner()` does indeed return an instance of `webkitAudioPannerNode`.

**Comment by d07RiV** [comment]
So it seems that webkitAudioPannerNode is a native function, but not a constructor.. that's odd. And if createPanner returns an instance of it, that means it's a 3D panner anyway, which is not quite as useful to me (I'd need to calculate displacement from relative volumes). I think it's easier to just disable stereo panning if the regular panner is not available.

Try 1.0.28.

**Comment by datenreisender** [comment]
No messages in the console anymore but also no sound.

**Comment by d07RiV** [comment]
Oh well then, I'm at a loss. They stopped developing Safari for windows ages ago, VM image is hard to find, browserstack does not support any modern Safari in the trial version. I think Apple intentionally doesn't want people to care about their browsers.

**Comment by datenreisender** [comment]
Sadly I currently also won't find the time to hack on this. Nevertheless, I think it is amazing, that with this project Diablo runs in a browser on iPads and iPhones at all, even if it is without sound!

Maybe just add a note to the `README.md`, that currently on Safari (desktop or mobile) no sound is played, until someone finds the time to implement it, so people do not expect it and know that with other desktop browsers they do get sound.

**Comment by OrangBuangan** [comment]
In case you weren't aware there is also a native iOS build of devilutionx where sound work on the iPhone:
https://github.com/diasurgical/devilutionX/pull/182

**Comment by Gamadril** [comment]
I can confirm that sounds work for me in Safari Version 12.1.2 (14607.3.9). MacOS Mojave 10.14.6

**Comment by Gamadril** [comment]
no, I have also problems with sound. Sometimes it works, but in most cases not. emptying cache, closing browser does not help. debbuger does not show any exceptions, the console is empty

**Comment by Gamadril** [comment]
I start thinking that this is a Safari bug. I cloned the repository and started playing with the sound.js file. Creating gainnode only at one place in play_sound (BTW why each sound saves it's own instance of it?), changing the node connection chain, removing the promise and saving the buffer directly in the sound instance. In some cases sound works only on the first load after rebuild. After page reload not any more.
So I reverted all my changes. Added an empty line forcing a rebuild and sound appears again, but always on the first page load. I also tried Safari 13 Beta - same problem. Seems the problem is sitting deep inside Apple's webkit implementation... (in v13 AudioContext is still prefixed as webkitAudioContext)
